File: tools/benchmarks/merlo/benchmarks/stage06p_extended/corpus/interface_closed_tag/python/main.py
import sys
MASK=(1<<64)-1
def apply(tag,value): return value*value if tag==0 else value+1
def run(n):
    operations=(0,1);total=0
    for i in range(n): total=(total+apply(operations[i&1],i))&MASK
    return total
print(run(int(sys.argv[1])))

File: tools/benchmarks/merlo/benchmarks/stage06p_extended/corpus/interface_closed_tag/go/main.go
package main
import("fmt";"os";"strconv")
type OperationTag uint8;const(Square OperationTag=iota;Increment);func apply(tag OperationTag,v uint64)uint64{if tag==Square{return v*v};return v+1}
func run(n uint64)uint64{ops:=[2]OperationTag{Square,Increment};var sum uint64;for i:=uint64(0);i<n;i++{sum+=apply(ops[i&1],i)};return sum}
func main(){n,e:=strconv.ParseUint(os.Args[1],10,64);if e!=nil{panic(e)};fmt.Printf("%d\n",run(n))}
